prometheus-server-metrics/src/main/scala/org/http4s/server/prometheus/PrometheusMetrics.scala

```diff
@@ -0,0 +1,197 @@
+package org.http4s.server.prometheus
+
+import io.prometheus.client._
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import fs2.Stream
+import org.http4s._
+import org.http4s.server.HttpMiddleware
+
+object PrometheusMetrics {
+
+  private case class SingleMetricRun(
+      activeRequests: Gauge,
+      totalRequests: Histogram.Timer,
+      requestTypeTimer: Histogram.Timer,
+      headers: Histogram.Timer,
+      abnormalTerminations: Counter,
+      serviceErrors: Counter
+  )
+
+  private case class GeneralServicMetrics(
+      activeRequests: Gauge,
+      totalRequests: Histogram,
+      headers: Histogram,
+      abnormalTerminations: Counter,
+      serviceErrors: Counter
+  )
+
+  private case class RequestHistograms(
+      getReq: Histogram,
+      postReq: Histogram,
+      putReq: Histogram,
+      headReq: Histogram,
+      moveReq: Histogram,
+      optionsReq: Histogram,
+      traceReq: Histogram,
+      connectReq: Histogram,
+      deleteReq: Histogram,
+      otherReq: Histogram
+  )
+
+  private case class ServiceMetrics(
+      generalMetrics: GeneralServicMetrics,
+      requestHist: RequestHistograms
+  )
+
+  private def metricsService[F[_]: Sync](
+      singleMetricRun: SingleMetricRun,
+      service: HttpService[F]
+  )(
+      req: Request[F]
+  ): OptionT[F, Response[F]] = OptionT {
+    for {
+      responseAtt <- service(req).value.attempt
+      _ <- Sync[F].delay(singleMetricRun.headers.observeDuration)
+      respOpt <- metricsServiceHandler[F](singleMetricRun, responseAtt)
+    } yield respOpt
+  }
+
+  private def metricsServiceHandler[F[_]: Sync](
+      s: SingleMetricRun,
+      e: Either[Throwable, Option[Response[F]]]
+  ): F[Option[Response[F]]] =
+    e.fold(
+      e => onServiceError(s) *> Sync[F].raiseError[Option[Response[F]]](e),
```

Comment:
noob question, unrelated to this PR, sorry... what is `*>`?

Reply:
Run both effect, but discard output of the left one. Same outcome as `>>`